src/main/java/com/synopsys/integration/alert/web/api/custom/CustomEndpointController.java

```diff
@@ -66,7 +70,18 @@ public CustomEndpointController(CustomEndpointManager customEndpointManager, Res
         }
 
         HttpServletContentWrapper servletContentWrapper = new HttpServletContentWrapper(httpRequest, httpResponse);
-        return customEndpointManager.performFunction(key, restModel, servletContentWrapper);
+        ActionResult<?> result = customEndpointManager.performFunction(key, restModel, servletContentWrapper);
+        if (result.isOk()) {
+            if (result.hasContent()) {
+                String content = gson.toJson(result.getContent().get());
+                return responseFactory.createOkContentResponse(content);
+            } else {
+                String message = result.getMessage().orElse("");
+                return responseFactory.createOkResponse("", message);
+            }
+        } else {
+            String message = result.getMessage().orElse("");
+            return responseFactory.createResponse(result.getHttpStatus(), null, message);
+        }
```

Comment:
This is just the first step and low hanging fruit to start implementing and changing the code.  Ultimately what I would like is similar to what you suggest.  I'm simply starting the process.  I actually need to change the branch to the feature branch because I have a feature branch where these changes will go.

Reply:
If the code is completely working, I would say skip the feature branch. I think we could all benefit from having the `ActionResult` object ASAP so that we don't need to add more tech-debt around this. 

Then I think a fast-follow MR fixing custom endpoints would be beneficial.